packages/smooth_app/lib/pages/product/product_page/header/product_tabs.dart

```diff
@@ -0,0 +1,239 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/preferences/user_preferences.dart';
+import 'package:smooth_app/generic_lib/buttons/smooth_simple_button.dart';
+import 'package:smooth_app/knowledge_panel/knowledge_panels_builder.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_card.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_dev_mode.dart';
+import 'package:smooth_app/pages/prices/prices_card.dart';
+import 'package:smooth_app/pages/product/product_page/header/reorder_bottom_sheet.dart';
+import 'package:smooth_app/pages/product/website_card.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+
+class ProductTab {
+  ProductTab({
+    required this.labelBuilder,
+    required this.builder,
+  });
+
+  final Widget Function(BuildContext) labelBuilder;
+  final Widget Function(Product) builder;
+}
+
+class ProductTabBar extends StatelessWidget {
+  const ProductTabBar({
+    required this.tabController,
+    required this.tabs,
+  });
+
+  final TabController tabController;
+  final List<ProductTab> tabs;
+
+  @override
+  Widget build(BuildContext context) {
+    final SmoothColorsThemeExtension themeExtension =
+        context.extension<SmoothColorsThemeExtension>();
+
+    return SliverPersistentHeader(
+      delegate: TabBarDelegate(
+        TabBar(
+          controller: tabController,
+          isScrollable: true,
+          tabs: tabs
+              .map((ProductTab tab) => Tab(
+                    child: tab.labelBuilder(context),
+                  ))
+              .toList(growable: false),
+          labelColor: Theme.of(context).primaryColor,
```

Comment:
When you call `Theme.of(context)` multiple times, don't forget that you have to find the widget in the hierarchy X times. So make it a var